### PR TITLE
feature: list processes in guest endpoint

### PR DESCRIPTION
**Pull request by ireznice**

### CODE REVIEW

File: lib/lab_manager/app/endpoints/computes.rb [hunk 1]
```diff
@@ -57,6 +57,26 @@ class Compute < Base
...
+          # everything is asynchronous
+          compute = ::Compute.find(params[:id])
+
+          halt 422, {
+            message: 'param user must be given'
+          }.to_json if params[:user].blank?
+
+          halt 422, {
+            message: 'param password must be given'
+          }.to_json if params[:password].blank?
```
Comment (lksv) [inline lib/lab_manager/app/endpoints/computes.rb]: clould be password empty string? I thing so.
Use `params[:password].nil?`

File: lib/lab_manager/workers/action_worker.rb [hunk 2]
```diff
@@ -211,6 +213,21 @@ def revert_snapshot_vm
       end
     end
 
+    def processes_vm
+      fail 'Compute has to be in running state' unless compute.state == 'running'
+      fail 'Wrong action payload' unless action.payload
```
Comment (lksv) [inline lib/lab_manager/workers/action_worker.rb]: `Hash === action.payload`

File: spec/integration/endpoints/computes_spec.rb [hunk 3]
```diff
@@ -363,4 +363,28 @@ def app
...
+  describe 'GET /computes/:compute_id/processes' do
+    let(:compute) { create(:compute, provider_name: 'v_sphere', name: 'one') }
+
+    it 'returns 404 when compute not found' do
+      get "/computes/#{compute.id + 200}/processes"
+      expect(last_response.status).to eq 404
+    end
+
+    it 'returns 404 when user not specified' do
+      get "/computes/#{compute.id}/processes", user: 'fooo'
```
Comment (lksv) [inline spec/integration/endpoints/computes_spec.rb]: you specifi "user" here.

**Comment by lksv** [comment]
I also prefer to set custom exception. For instance in action you use just `fail`.

**Comment by ireznice** [comment]
@lksv fixed

**Pull request merged**